source/nodejs/adaptivecards-templating/src/template-engine.ts

```diff
@@ -49,6 +53,23 @@ class EvaluationContext {
     set $data(value: any) {
         this._$data = value;
     }
+
+    generateVersionJson() {
+        const version = pkg.version;
+        const versionSplit = version.split('.');
+
+        let patchSplit = [];
+        if (versionSplit[2]) {
+            patchSplit = versionSplit[2].split('-');
+        }
+
+        return {
+            "major": parseInt(versionSplit[0]),
```

Comment:
nit: can you add a check with a regex to ensure the whole of input string for the version is well-formed?

Reply:
Is it necessary to check with a regex since we set the version ourselves?

Reply:
sure, it's reasonable. but don't you want to verify that input string is in the correct order? we are assuming that input string is in the certain format and access them accordingly. 
we can have a new valid versioning format that may not confirm to this assumption in the future. 
if you have a regex check, when regex fails, other maintainers will be able to see what's expected format & quickly reason about it.
and regex check ensures that we don't get weird errors we might get from something like 'xxx.xxx.xxx.xxx' 
as i've stated, it's nit.

Reply:
if you create a regex, not only that you can check the validity of the input, but you can also extract the value using it; just make sure you group them using '()'

Reply:
I think the existing logic is enough:
Keep it simple, and make it crash/test case can detect it if the version doesn't meet 2.3.0-alpha format.


nodejs version rule is a little complex and we don't need to support all of them.
https://docs.npmjs.com/cli/v6/configuring-npm/package-json#version
https://github.com/isaacs/node-semver#usage

Reply:
Thanks for all of the versioning info! We test to make sure major, minor and patch are always parsed correctly. The only one that's a little funny is the suffix since it's optional on our end